samples/interactive-tutorials/setup/update-user-events-json.js

```diff
@@ -23,8 +23,18 @@ async function main() {
 
     for (let i = 0; i < events.length - 1; ++i) {
       const event = JSON.parse(`[${events[i]}]`)[0];
-      const date = new Date(event.eventTime);
-      date.setDate(date.getDate() - 1);
+      let date = new Date(event.eventTime);
+      const nowDate = new Date();
+      const timeDiff = nowDate - date;
+
+      if (timeDiff > 1000 * 60 * 60 * 24 * 90) {
+        if (nowDate.getDate() - 1 === 0) {
+          const yesterday = nowDate - 1000 * 60 * 60 * 24;
+          date = new Date(yesterday);
+        } else {
+          date = nowDate.getDate() - 1;
```

Comment:
This does not make sense to me. Line 35 will assign an integer number to `date`.

Why do you need to check that `now.getDate() - 1 === 0`? Even today, on August 1st, the old code would work:

```ts
> d = new Date()
2022-08-01T16:57:37.463Z
> d.setDate(d.getDate() - 1)
1659286657463
> d
2022-07-31T16:57:37.463Z
```

Reply:
The previous version of that code didn't work correctly, as it should change eventTime if it is earlier than 90 days, it should change the eventTime to yesterday, but when i changed it with this code now.getDate() - 1, it returned me 0, that's why i added that if statement for check.